### PR TITLE
qa/tasks/quiescer: dump ops in parallel

### PR DESCRIPTION
**Pull request by batrick**
Since this --flags=locks takes the mds_lock and dumps thousands of ops, this may take a long time to complete for each individual MDS. The entire quiesce set may timeout (and all q ops killed) before we finish dumping ops.

Fixes: https://tracker.ceph.com/issues/65823

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
- Component impact
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by batrick** [comment]
Not yet tested.

File: qa/tasks/quiescer.py [hunk 1]
```diff
@@ -173,6 +173,7 @@ def dump_ops_all_ranks(self, dump_tag):
         remote_dumps = []
 
         # begin by executing dump on all ranks
+        procs = []
```
Comment (leonid-s-usov) [inline qa/tasks/quiescer.py:176]: we already have the `remote_dumps` array, let's just put the process info there

File: qa/tasks/quiescer.py [hunk 2]
```diff
@@ -186,11 +187,15 @@ def dump_ops_all_ranks(self, dump_tag):
...
+                procs.append(p)
                 remote_dumps.append((info, remote_path))
             except Exception as e:
                 self.logger.error(f"Couldn't execute ops dump on rank {rank}, error: {e}")
 
+        for p in procs:
+            p.wait()
+
         # now get the ops from the files
         for info, remote_path in remote_dumps:
```
Comment (leonid-s-usov) [inline qa/tasks/quiescer.py:200]: and then insert the `p.wait` call in this loop under the `try:` to collect at least some of the dumps if one fails

**Review by leonid-s-usov — CHANGES_REQUESTED** [review]
Thanks for submitting this, Patrick!

File: qa/tasks/quiescer.py (hunk 2 above)
Comment (batrick) [reply qa/tasks/quiescer.py]: done

File: qa/tasks/quiescer.py (hunk 1 above)
Comment (batrick) [reply qa/tasks/quiescer.py]: done

**Review by leonid-s-usov — APPROVED** [review]
Looks good! It should be easy to test this with any of the upcoming teuthology batches by using this branch as the suite

**Comment by batrick** [comment]
jenkins test api

**Comment by batrick** [comment]
jenkins test make check arm64

**Comment by batrick** [comment]
jenkins test make check arm64

**Comment by batrick** [comment]
jenkins test make check arm64

**Comment by batrick** [comment]
This PR is under test in [https://tracker.ceph.com/issues/65867](https://tracker.ceph.com/issues/65867).

**Comment by batrick** [comment]
jenkins test make check arm64

File: qa/tasks/quiescer.py [hunk 3]
```diff
@@ -186,14 +186,15 @@ def dump_ops_all_ranks(self, dump_tag):
 
             self.logger.debug(f"Dumping ops on rank {rank} ({name}) to a remote file {remote_path}")
             try:
-                _ = self.fs.rank_tell(['ops', '--flags=locks', f'--path={daemon_path}'], rank=rank)
-                remote_dumps.append((info, remote_path))
+                p = self.fs.rank_tell(['ops', '--flags=locks', f'--path={daemon_path}'], rank=rank, wait=False)
```
Comment (batrick) [inline qa/tasks/quiescer.py:189]: My mistake, I cannot do this:

https://pulpito.ceph.com/pdonnell-2024-05-08_22:06:20-fs-wip-pdonnell-testing-20240508.183908-debug-distro-default-smithi/7698868/

```
2024-05-09T00:35:43.460 ERROR:tasks.quiescer.fs.[cephfs]:Couldn't pull ops dump at '/var/run/ceph/b96c13bc-0d98-11ef-bc97-c7b262605968/ops-7749c26b-1-mds.i.json' on rank 2 (i), error: 'dict' object has no attribute 'wait'
```

Comment (leonid-s-usov) [reply qa/tasks/quiescer.py:189]: :'( I was looking forward to the PR... How hard is it to add the async capability to the `rank_tell`?

Comment (batrick) [reply qa/tasks/quiescer.py:189]: Instead of using the helper `rank_tell`, probably just manually submit the command instead.

**Comment by batrick** [comment]
https://pulpito.ceph.com/pdonnell-2024-05-16_16:19:21-fs:workload-main-distro-default-smithi/

**Comment by batrick** [comment]
jenkins test make check arm64

**Comment by batrick** [comment]
> https://pulpito.ceph.com/pdonnell-2024-05-16_16:19:21-fs:workload-main-distro-default-smithi/

seems to work as advertised now: /teuthology/pdonnell-2024-05-16_16:19:21-fs:workload-main-distro-default-smithi/7709343/teuthology.log

**Pull request merged**